src/finalFrame.js

```diff
@@ -0,0 +1,69 @@
+function FinalFrame() {
+  this.rolls = [];
+  this.strikes = [false, false];
+  this.spare = false;
+  this.remaining = 10;
+}
+
+FinalFrame.prototype.roll = function roll(score) {
+  if (this.over === true) {
+    throw new Error('The frame is already over.');
+  } else if (score > 10) {
+    throw new Error('The score on one roll cannot be over 10.');
+  } else if (score > this.remaining) {
+    throw new Error('You cannot knock down more pins than there are standing.');
+  }
+  this.scoreRoll(score);
+};
+
+FinalFrame.prototype.scoreRoll = function scoreRoll(score) {
+  this.rolls.push(score);
+  if (this.rolls.length === 3) {
+    this.over = true;
```

Comment:
If you have state, you should declare it in your constructure. Does this need to be state though? Could it not be a method, e.g. 

```
function isOver() { 
  return this.rolls.length >= 3 || this.rolls.length >= 2 && this.score() < 10;
}
```

Generally unless we really care about performance, we want to keen this stuff dynamic, as then we know that it's properly reflective of the state. Otherwise we could introduce more complexity, in that we have to check that the this.over state is maintained alongside the score.